### ISSUE TITLE
Uncaught (in promise) Error in browser console

### ISSUE THREAD

**Issue by larrxu** [apollostack/react-apollo#404]
## Steps to Reproduce

1. Start the Githunt-API and -React projects
2. Load the first page, without login.
3. Change Githunt-API's Query schema to `currentUser: User!` (making it a non-null field).
4. Go to browser and click any link or wait for hot reload.
5. Uncaught (in promise) Error is shown in console.

### Buggy Behavior

The error is shown as a uncaught promise error. There doesn't seem a way to stop/catch it.

### Expected Behavior

I should be able to handle the error using data.error in my React component, resulting a cleaner error handling. Site viewers should never see the red error message in browser console. Or, I should be able to pass in an option leading to the error not being thrown.

```javascript
export default graphql(PROFILE_QUERY, {
  options: { forceFetch: true },
  props: ({ data: { loading, error, currentUser } }) => ({
    loading, error, currentUser,
  }),
})(Profile);
```

### Version
- apollo-client@0.6.0
- react-apollo@0.8.1

If this is by design, could you advice how to get rid of the error message in console?

**Comment by tmeasday** [comment]
You can see the error in the `error` prop on `data` (note that in this case the `currentUser` is also requested by the `FeedPage`.

However:

a) there doesn't indeed seem to be a way to avoid the error going to the console.
b) this actually completely breaks SSR.

**Comment by larrxu** [comment]
Thank you for looking into this. For SSR I actually created an issue on Githunt earlier.
https://github.com/apollostack/GitHunt-React/issues/180

**Comment by tmeasday** [comment]
Fixed by #408
